builtin/logical/pki/cert_util.go

```diff
@@ -1240,6 +1259,9 @@ func signCertificate(data *dataBundle) (*certutil.ParsedCertBundle, error) {
 		if certTemplate.MaxPathLen == 0 {
 			certTemplate.MaxPathLenZero = true
 		}
+	} else {
+		certTemplate.BasicConstraintsValid = true
```

Comment:
What's the reason for this change? Also, IsCA wouldn't need to be set explicitly false, and if BasicConstraintsValid should be true in either case it should be pulled out of the if condition.

Reply:
client/server certificates should have this constraint explicitly set. Some old tools/browser refuse to connect to web servers with a certificate without the basicConstraint CA:False

To not change the default behaviour, I updated the patch to add an option in the pki role for that.